src/storage/heap_file.c

```diff
@@ -7051,12 +7035,9 @@ heap_scancache_quick_end (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache)
     }
   else
     {
-      if (scan_cache->index_stat_info != NULL)
-	{
-	  /* deallocate memory space allocated for index stat info. */
-	  db_private_free_and_init (thread_p, scan_cache->index_stat_info);
-	  scan_cache->num_btids = 0;
-	}
+      delete scan_cache->m_index_stats;
```

Comment:
if we delete here we should add #include "btree_unique.hpp" to be sure we won't have incomplete type deletes in the future

Reply:
fixed